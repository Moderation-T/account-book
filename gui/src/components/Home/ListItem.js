import React from 'react';
import { Row, Col } from 'antd';
import { EditOutlined, DeleteOutlined } from '@ant-design/icons';
import PropTypes from 'prop-types';

function ListItem({ item, onModifyItem, onDeleteItem }) {
  // 判断收入还是支出
  // const type = item.category.type === 'outcome' ? '-' : '+';

  return (
    <Row gutter={16}>
      <Col span={4} className="title">
        {item.title}
      </Col>
      <Col span={4} className="price">
        {/* {type} */}
        {item.price}
      </Col>
      <Col span={4} className="date">
        {item.date}
      </Col>
      <Col span={4} className="edit">
        <EditOutlined onClick={onModifyItem} />
      </Col>
      <Col span={4} className="delete">
        <DeleteOutlined onClick={onDeleteItem} />
      </Col>
    </Row>
  );
}

// 类型检查
ListItem.protoTypes = {
  // 函数类型且是必须的
  item: PropTypes.object.isRequired,
  onDeleteItem: PropTypes.func.isRequired,
  onModifyItem: PropTypes.func.isRequired,
};

/* 
// 可以给组件属性设置默认值
ListItem.defaultProps = {
  // 函数类型且是必须的
  onModifyItem: () => {},
};
 */

export default ListItem;
