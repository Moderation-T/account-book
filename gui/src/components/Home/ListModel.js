import React from 'react'
import {Tabs} from 'antd'

import ListItem from './ListItem'

const {TabPane} = Tabs;

function ListModel(params) {
  return   <ListItem></ListItem>
}

export default ListModel;