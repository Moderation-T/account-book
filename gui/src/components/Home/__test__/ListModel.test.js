import React from 'react';
import { shallow } from 'enzyme';

import ListItem from '../ListItem';
import ListModel from '../ListModel';

/** 测试用例说明
 * 1. 判断 LIstItem 渲染长度是否符合预期
 */

const items = [
  {
    title: 'buy stuff for kitten',
    price: 100,
    date: '2018-08-15',
    monthCategory: '2018-8',
    id: '_kly1klf4g',
    cid: '1',
    timestamp: 1534291200000,
  },
  {
    title: '这是我的工资',
    price: 20000,
    date: '2018-08-18',
    monthCategory: '2018-8',
    id: '_bd16bjeen',
    cid: '2',
    timestamp: 1534550400000,
  },
];

const props = { items, onModifyItem: jest.fn(), onDeleteItem: jest.fn() };

let wrapper;
describe('test ListModel component', () => {
  //  每次运行单个测试用例的时候会执行这一步
  beforeEach(() => {
    wrapper = shallow(<ListModel {...props}></ListModel>);
  });

  it('should render the component to match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correct array length', () => {
    expect(wrapper.find(ListItem).length).toEqual(items.length);
  });
});
