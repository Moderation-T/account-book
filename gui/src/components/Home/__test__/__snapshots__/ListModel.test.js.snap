// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test ListModel component should render the component to match snapshot 1`] = `ShallowWrapper {}`;
